lib/coa/utils/concourse/fly_client.rb

```diff
@@ -0,0 +1,82 @@
+require_relative '../command_runner'
+
+module Coa
+  module Utils
+    module Concourse
+      # This class serves as an interface to Concourse's CLI "fly".
+      class FlyClient
+        attr_reader :target, :creds
+
+        def initialize(target, creds)
+          @target = target
+          @creds = creds
+        end
+
+        def login
```

Comment:
maybe we can move this into `initialize` to have an `auto-login` feature, it will be more developper friendly as it won't require a first call to login before (see http://www.changit.fr/bug-free-by-design/#25)

Reply:
It was like this originally and I thought about it a lot and did not like the idea of doing this process at the object creation so I moved it out. Which was the reason why I decoupled the creation of concourse's fly_client outside of the "concourse client" object creation. I'll have a look at your link and we can talk about it.